websubhub-ballerina/http_service.bal

```diff
@@ -18,56 +18,40 @@ import ballerina/http;
+    private final boolean isRegisterAvailable;
+    private final boolean isDeregisterAvailable;
 
     # Initializes the `websubhub:HttpService` endpoint.
     # ```ballerina
-    # websubhub:HttpService httpServiceEp = check new ('service, "https://sample.hub.com", 3600);
+    # websubhub:HttpService httpServiceEp = check new (handler, "https://sample.hub.com", 3600);
     # ```
     #
-    # + hubService   - Current `websubhub:Service` instance
+    # + handler - The `websubhub:RequestHandler` instance which used as a wrapper to execute service methods
     # + hubUrl       - Hub URL
     # + leaseSeconds - Subscription expiration time for the `hub`
     # + clientConfig - The `websubhub:ClientConfiguration` to be used in the HTTP Client used for subscription/unsubscription intent verification
     # + return - The `websubhub:HttpService` or an `error` if the initialization failed
-    public isolated function init(Service hubService, string hubUrl, int leaseSeconds, *ClientConfiguration clientConfig) {
-        self.hubService = hubService;
-        self.clientConfig = clientConfig;
+    isolated function init(RequestHandler handler, string hubUrl, int leaseSeconds, 
+                                  string[] methodNames, *ClientConfiguration clientConfig) {
+        self.handler = handler;
+        self.clientConfig = clientConfig.cloneReadOnly();
         self.hub = hubUrl;
         self.defaultHubLeaseSeconds = leaseSeconds;
-        string[] methodNames = getServiceMethodNames(hubService);
-        foreach var methodName in methodNames {
-            match methodName {
-                "onSubscription" => {
-                    self.isSubscriptionAvailable = true;  
-                }
-                "onSubscriptionValidation" => {
-                    self.isSubscriptionValidationAvailable = true;
-                }
-                "onUnsubscription" => {
-                    self.isUnsubscriptionAvailable = true;
-                }
-                "onUnsubscriptionValidation" => {
-                    self.isUnsubscriptionValidationAvailable = true;
-                }
-                "onRegisterTopic" => {
-                    self.isRegisterAvailable = true;
-                }
-                "onDeregisterTopic" => {
-                    self.isDeregisterAvailable = true;
-                }
-            }
-        }
+        self.isSubscriptionAvailable = isMethodAvailable("onSubscription", methodNames);
+        self.isSubscriptionValidationAvailable = isMethodAvailable("onSubscriptionValidation", methodNames);
+        self.isUnsubscriptionAvailable = isMethodAvailable("onUnsubscription", methodNames);
+        self.isUnsubscriptionValidationAvailable = isMethodAvailable("onUnsubscriptionValidation", methodNames);
+        self.isRegisterAvailable = isMethodAvailable("onRegisterTopic", methodNames);
+        self.isDeregisterAvailable = isMethodAvailable("onDeregisterTopic", methodNames);
```

Comment:
Any reason for this change ?

Reply:
* Resolving since we have concluded to use this in our discussion.